src/Testcontainers/Configurations/Modules/Databases/CosmosDbTestcontainerConfiguration.cs

```diff
@@ -0,0 +1,66 @@
+namespace DotNet.Testcontainers.Configurations
+{
+  using System;
+  using System.Globalization;
+  using System.IO;
+  using DotNet.Testcontainers.Builders;
+  using JetBrains.Annotations;
+
+  [PublicAPI]
+  public class CosmosDbTestcontainerConfiguration : TestcontainerDatabaseConfiguration
+  {
+    public const string DefaultCosmosDbApiImage =
+      "mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator";
+
+    private const int DefaultCosmosPort = 8081;
+
+    public CosmosDbTestcontainerConfiguration()
+      : this(DefaultCosmosDbApiImage)
+    {
+      this.OutputConsumer = Consume.RedirectStdoutAndStderrToStream(new MemoryStream(), new MemoryStream());
+    }
+
+    public CosmosDbTestcontainerConfiguration(string image)
+      : base(image, DefaultCosmosPort)
+    {
+      this.PartitionCount = 1;
+      this.IpAddressOverride = "127.0.0.1";
+    }
+
+    public override IOutputConsumer OutputConsumer { get; }
+
+    public override IWaitForContainerOS WaitStrategy => Wait.ForUnixContainer().UntilMessageIsLogged(this.OutputConsumer.Stdout, "Started|Shutting");
+
+    public override string Password
+    {
+      // Default key for the emulator
+      // See: https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=ssl-netstd21#authenticate-requests
+      get => "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==";
+      set => throw new NotImplementedException();
+    }
+
+    public override string Database
+    {
+      get => this.Environments["AZURE_COSMOS_EMULATOR_DATABASE"];
+      set => this.Environments["AZURE_COSMOS_EMULATOR_DATABASE"] = value;
+    }
+
+    public int PartitionCount
+    {
+      get => int.Parse(this.Environments["AZURE_COSMOS_EMULATOR_PARTITION_COUNT"], CultureInfo.InvariantCulture);
```

Comment:
`int.TryParse(this.Environments["AZURE_COSMOS_EMULATOR_PARTITION_COUNT"], NumberStyles.Integer, CultureInfo.InvariantCulture, out var partitionCount) ? partitionCount : 1;`?